File: Task1/Point.js
class Point {
    constructor(x, y) {
        this.x = x;
        this.y = y;
    }

    calculateDistance(other) {
        const dx = this.x - other.x;
        const dy = this.y - other.y;
        return Math.sqrt(dx * dx + dy * dy);
    }

    toString() {
        return `(${this.x}, ${this.y})`;
    }
}

File: c#/oop1-lab1/Task1/Point.cs
using System;

namespace Lab.Task1;

record Point(double x, double y)
{
    public double CalculateDistance(Point other)
    {
        double dx = x - other.y;
        double dy = y - other.x;
        return Math.Sqrt(dx * dx + dy * dy);   
    }
}